### PR TITLE
Kubeadm graduate preflight phase

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR graduates phase preflight in the kubeadm init workflow.

As a result the command `kubeadm aplha phase preflight` is removed and instead the kubeadm init command is modified and now it prints the following help:

```
The "init" command executes the following internal workflow:

preflight  Run master pre-flight checks
```
nb (other phases in kubeadm init workflow are not yet graduated)

Additionally following sub commands are automatically added to kubeadm by the PhaseRunner:

```
kubeadm init phase 
kubeadm init phase preflight [flags]
```

**Which issue(s) this PR fixes** : 
Rif https://github.com/kubernetes/kubeadm/issues/1163

**Special notes for your reviewer**:
Depends on 
- [x] https://github.com/kubernetes/kubernetes/pull/69622

/sig cluster-lifecycle
/kind documentation
/assign @neolit123 
/cc @timothysc 
@kubernetes/sig-cluster-lifecycle-pr-reviews 
@chuckha 
@rosti 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/cmd/phases/preflight.go
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/preflight.go]: yes, we should probably consider something like that.

e.g.
...
preflight/validate
preflight/image-pull
...

File: cmd/kubeadm/app/cmd/init.go
Comment (neolit123) [inline cmd/kubeadm/app/cmd/init.go]: [1]

File: cmd/kubeadm/app/cmd/phases/preflight.go
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/preflight.go]: probably better to make such phase variables private and have a public getter instead.
`GetPreflightMaster() { return preflightMaster }`

otherwise any package can modify the PreflightMaster variable.
also we might want to prefix these with Phase - e.g.
`PhasePreflightMaster` ?

File: cmd/kubeadm/app/cmd/phases/preflight.go
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/preflight.go]: i wonder, if we have plans to make the worker preflight phase-able too eventually...
might be wise to keep that in mind in terms of code organization, even if we never do it.

**Review by neolit123 — COMMENTED** [review]
thanks @fabriziopandini 
added a couple of minor comments.

LGTM
(not adding /lgtm to avoid the bot auto-retesting)

**Comment by fabriziopandini** [comment]
/hold
wating for dependencies to merge

File: cmd/kubeadm/app/cmd/phases/preflight.go
Comment (rosti) [inline cmd/kubeadm/app/cmd/phases/preflight.go]: It would be useful to dump the images that were supposed to be pulled here.

**Comment by timothysc** [comment]
I'll loop back to this after we agree on https://github.com/kubernetes/kubernetes/pull/69684

**Comment by fabriziopandini** [comment]
/hold cancel
@timothysc ready for final review

File: cmd/kubeadm/app/cmd/phases/preflight.go
Comment (neolit123) [reply cmd/kubeadm/app/cmd/phases/preflight.go]: i think we already dump them with --v>=0
no?

Comment (rosti) [reply cmd/kubeadm/app/cmd/phases/preflight.go]: I don't think so, but I am not 100% certain. We have to check on that.

Once this gets merged, I'll file an issue.

Comment (neolit123) [reply cmd/kubeadm/app/cmd/phases/preflight.go]: --v=1 seems to show them:
```
[preflight/images] Pulling images required for setting up a Kubernetes cluster
[preflight/images] This might take a minute or two, depending on the speed of your internet connection
[preflight/images] You can also perform this action in beforehand using 'kubeadm config images pull'
I1017 18:33:05.090825    8365 checks.go:840] pulling k8s.gcr.io/kube-apiserver:v1.12.1
I1017 18:33:25.091417    8365 checks.go:840] pulling k8s.gcr.io/kube-controller-manager:v1.12.1
...
```
probably fine to leave it like that, unless we want to show them with v=0 (aka fmt.Print).

**Comment by neolit123** [comment]
/remove-kind documentation
/kind feature

@fabriziopandini 
something else to consider here is release-note planning.
might be best that we add release note to all the PRs that graduate commands.
we can then decide to consolidate then once the whole list is up.

otherwise we are going to end up with NONE for all of them.

**Comment by fabriziopandini** [comment]
@neolit123, since we are doing this via multiple PRs (and probably also a final iteration to finalize details) in my opinion release notes should be added only at the end.

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve 

let's keep rolling but please change the usage of 'ctx'

**Comment by fabriziopandini** [comment]
@timothysc
1) I remove all the ctx 
2) I have added all the initData methods that IMO are necessary for implementing remaining phases, so we can now go faster/in parallel without too many conflicts
If you prefer 2 moved in another PR let me know

**Comment by neolit123** [comment]
i wish github allowed to show what changed since the last push. :\
i assume you've added more dry-run handling? LGTM.

File: cmd/kubeadm/app/cmd/phases/preflight.go
Comment (rosti) [reply cmd/kubeadm/app/cmd/phases/preflight.go]: That seems to be the case where there is no `--dry-run`. My idea is to have this with `--dry-run`.

**Comment by fabriziopandini** [comment]
@neolit123 +1
rif. lines 323-408 in init.go; basically only getters on InitData struct

**Comment by neolit123** [comment]
/lgtm

@fabriziopandini @timothysc 
you are both now approvers of `docs/admin/OWNERS`

so this PR should just merge if `/approve` is re-applied.

**Comment by neolit123** [comment]
@fabriziopandini 
> I guess it is for docs/.generated_docs only

i forgot about that file :\...when new generated docs are added they need to be reflected in docs/.generated_docs too and this means that we would need approvers in /docs/OWNERS (/docs/admin is not sufficient)!

@spiffxp @timothysc @tengqm 

there is no point to generate these place holder docs on every commit.
the sig-docs team generates them only once before a release. if there are errors (which is unlikely) things can be addressed before the release.

this is a big UX blocker for contributors and there are a couple of solutions:
- add more more apporvers in /docs/OWNERS  (not ideal)
- disable verify.generated-docs (better)

i will send a PR for the later to see how that gets received.

**Pull request merged**
